### ISSUE TITLE
drawDirectionField() can be wrong if add=FALSE

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#890]
This problem can arise because of a problem in checking `par` too early.

**Comment by dankelley** [comment]
A test code is 889.R, as below
```r
library(oce)
x <- 1:10
y <- 1:10
u <- rep(0.25, 10)
v <- u
dy <- 0.5

drawDirectionField(x, y, u, v, scalex=1, type=1)

drawDirectionField(x, y, u, v, scalex=1, type=1)
y <- y + dy
drawDirectionField(x, y, u, v, scalex=1, type=1, add=TRUE, length=0.1, col='red')
y <- y + dy
drawDirectionField(x, y, u, v, scalex=1, type=1, add=TRUE, length=0.025, col='blue')
y <- y + dy
drawDirectionField(x, y, u, v, scalex=1, type=2, add=TRUE)
y <- y + dy
drawDirectionField(x, y, u, v, scalex=1, type=2, add=TRUE, length=0.1, col='red')
y <- y + dy
drawDirectionField(x, y, u, v, scalex=1, type=2, add=TRUE, length=0.025, col='blue')
y <- y + dy

```

**Comment by dankelley** [comment]
Fixed in `develop` commit f8f65720adb4f5d7115dcdbb2c007ac6697d941a
